src/identity/mod.rs

```diff
@@ -37,9 +33,45 @@ impl PublicId {
     /// Returns the entity's network address.
     pub fn name(&self) -> &XorName {
         match self {
-            PublicId::Node(pub_id) => pub_id.name(),
-            PublicId::Client(pub_id) => pub_id.name(),
-            PublicId::App(pub_id) => pub_id.owner_name(),
+            Self::Node(pub_id) => pub_id.name(),
+            Self::Client(pub_id) => pub_id.name(),
+            Self::App(pub_id) => pub_id.owner_name(),
+        }
+    }
+
+    /// Returns the node public id, if applicable.
+    pub fn node_public_id(&self) -> Option<&node::PublicId> {
+        if let Self::Node(id) = self {
+            Some(id)
+        } else {
+            None
+        }
+    }
+
+    /// Returns the client public id, if applicable.
+    pub fn client_public_id(&self) -> Option<&client::PublicId> {
+        if let Self::Client(id) = self {
+            Some(id)
+        } else {
+            None
+        }
+    }
+
+    /// Returns the app public id, if applicable.
+    pub fn app_public_id(&self) -> Option<&app::PublicId> {
+        if let Self::App(id) = self {
+            Some(id)
+        } else {
+            None
+        }
+    }
+
+    /// Returns the entity's public key, if applicable.
+    pub fn public_key(&self) -> Option<PublicKey> {
```

Comment:
Can this function return just `PublicKey` now?
(Clippy has also failed)

Reply:
The failures on Jenkins are due to an out of date container. However, I think @jacderida said that the container fails to build -- seems like it can't be updated without the lint allowance in this PR. I would say let's ignore Jenkins for this PR.

Reply:
That's cool then! :+1: